xbmc/video/VideoItemArtworkHandler.cpp

```diff
@@ -0,0 +1,472 @@
+
+  if (currentArt.empty())
+  {
+    CVideoDatabase videodb;
+    if (!videodb.Open())
+    {
+      CLog::LogF(LOGERROR, "Cannot open video database!");
+      return {};
+    }
+
+    currentArt = videodb.GetArtForItem(m_item->GetVideoInfoTag()->m_iDbId,
+                                       m_item->GetVideoInfoTag()->m_type, "thumb");
+  }
+  return currentArt;
+}
+
+std::vector<std::string> CVideoItemArtworkArtistHandler::GetRemoteArt()
+{
+  return {};
+}
+
+std::string CVideoItemArtworkArtistHandler::GetLocalArt()
+{
+  CMusicDatabase musicdb;
+  if (!musicdb.Open())
+  {
+    CLog::LogF(LOGERROR, "Cannot open music database!");
+    return {};
+  }
+
+  std::string localArt;
+  const int idArtist = musicdb.GetArtistByName(m_item->GetLabel());
+  if (idArtist >= 0)
+  {
+    // Get artist paths - possible locations for thumb - while music db open
+    CArtist artist;
+    musicdb.GetArtist(idArtist, artist);
+    std::string artistPath;
+    musicdb.GetArtistPath(artist, artistPath); // Artist path in artist info folder
+
+    std::string thumb;
+    bool existsThumb = false;
+
+    // First look for artist thumb in the primary location
+    if (!artistPath.empty())
+    {
+      thumb = URIUtils::AddFileToFolder(artistPath, "folder.jpg");
+      existsThumb = CFileUtils::Exists(thumb);
+    }
+    // If not there fall back local to music files (historic location for those album artists with a unique folder)
+    if (!existsThumb)
+    {
+      std::string artistOldPath;
+      musicdb.GetOldArtistPath(idArtist, artistOldPath); // Old artist path, local to music files
+      if (!artistOldPath.empty())
+      {
+        thumb = URIUtils::AddFileToFolder(artistOldPath, "folder.jpg");
+        existsThumb = CFileUtils::Exists(thumb);
+      }
+    }
+
+    if (existsThumb)
+      localArt = thumb;
+  }
+  return localArt;
+}
+
+void CVideoItemArtworkArtistHandler::PersistArt(const std::string& art)
+{
+  CMusicDatabase musicdb;
+  if (!musicdb.Open())
+  {
+    CLog::LogF(LOGERROR, "Cannot open music database!");
+    return;
+  }
+
+  const int idArtist = musicdb.GetArtistByName(m_item->GetLabel());
+  if (idArtist >= 0)
+    musicdb.SetArtForItem(idArtist, MediaTypeArtist, m_artType, art);
+}
+
+//-------------------------------------------------------------------------------------------------
+// CVideoItemArtworkActorHandler (Actor handler)
+//-------------------------------------------------------------------------------------------------
+
+class CVideoItemArtworkActorHandler : public CVideoItemArtworkHandler
+{
+public:
+  explicit CVideoItemArtworkActorHandler(const std::shared_ptr<CFileItem>& item,
+                                         const std::string& artType)
+    : CVideoItemArtworkHandler(item, artType)
+  {
+  }
+
+  std::string GetCurrentArt() override;
+  std::string GetLocalArt() override;
+
+  std::string GetDefaultIcon() override { return "DefaultActor.png"; }
```

Comment:
const?

Reply:
I never would make interface defined methods `const` as you define on the interface level implementation constraints and may force actual implementations that need for example lazy evaluation to introduce `const` casts, `mutable`members and what not. Of course this is debatable.

Reply:
Okay, revising my own question, you are right, "const as const can". Will change.

Reply:
Yeah, I guess you were referring to the returned types and not really to the const correctness of the method. Getters should not be able to change internals (apart from locks/mutex that can be made mutable anyway)

Reply:
> (apart from locks/mutex that can be made mutable anyway)

... and mutable variables used to cache (lazy evaluated) values.